presto-iceberg/src/test/java/io/prestosql/plugin/iceberg/TestIcebergMetadataListing.java

```diff
@@ -0,0 +1,197 @@
+    public void setUp()
+    {
+        @Language("SQL") String createSchema = "CREATE SCHEMA "
+                + HIVE_CATALOG + "." + TEST_SCHEMA;
+
+        @Language("SQL") String createIcebergTable1 = "CREATE TABLE "
+                + ICEBERG_CATALOG + "." + TEST_SCHEMA + "." + ICEBERG_TABLE1
+                + "("
+                + "  _string VARCHAR"
+                + ", _integer INTEGER"
+                + ")";
+
+        @Language("SQL") String createIcebergTable2 = "CREATE TABLE "
+                + ICEBERG_CATALOG + "." + TEST_SCHEMA + "." + ICEBERG_TABLE2
+                + "("
+                + "  _string VARCHAR"
+                + ", _integer INTEGER"
+                + ", _double DOUBLE"
+                + ", _boolean BOOLEAN"
+                + ") "
+                + "WITH ("
+                + "partitioning = ARRAY["
+                + "  '_string'"
+                + ", '_integer'])";
+
+        @Language("SQL") String createHiveTable = "CREATE TABLE "
+                + HIVE_CATALOG + "." + TEST_SCHEMA + "." + HIVE_TABLE
+                + "("
+                + "  _string VARCHAR"
+                + ", _integer INTEGER"
+                + ", _double DOUBLE"
+                + ")";
+
+        assertQuerySucceeds(createSchema);
+        assertQuerySucceeds(createIcebergTable1);
+        assertQuerySucceeds(createIcebergTable2);
+        assertQuerySucceeds(createHiveTable);
+    }
+
+    @AfterClass
+    public void tearDown()
+    {
+        @Language("SQL") String dropSchema = "DROP SCHEMA "
+                + HIVE_CATALOG + "." + TEST_SCHEMA;
+
+        @Language("SQL") String dropIcebergTable1 = "DROP TABLE "
+                + ICEBERG_CATALOG + "." + TEST_SCHEMA + "." + ICEBERG_TABLE1;
+
+        @Language("SQL") String dropIcebergTable2 = "DROP TABLE "
+                + ICEBERG_CATALOG + "." + TEST_SCHEMA + "." + ICEBERG_TABLE2;
+
+        @Language("SQL") String dropHiveTable = "DROP TABLE "
+                + HIVE_CATALOG + "." + TEST_SCHEMA + "." + HIVE_TABLE;
+
+        assertQuerySucceeds(dropIcebergTable1);
+        assertQuerySucceeds(dropIcebergTable2);
+        assertQuerySucceeds(dropHiveTable);
+        assertQuerySucceeds(dropSchema);
+    }
+
+    @Test
+    public void testTableListing()
+    {
+        @Language("SQL") String showIcebergTables = "SHOW TABLES FROM "
+                + ICEBERG_CATALOG + "." + TEST_SCHEMA;
+
+        @Language("SQL") String expected = "VALUES "
+                + "'" + ICEBERG_TABLE1 + "'"
+                + ", '" + ICEBERG_TABLE2 + "'";
+
+        assertQuery(showIcebergTables, expected);
+    }
+
+    @Test
+    public void testTableColumnListing()
+    {
+        @Language("SQL") String countIcebergTableColumns = "SELECT count(*) FROM "
+                + ICEBERG_CATALOG + "." + INFORMATION_SCHEMA + "." + TABLE_COLUMNS.getTableName()
+                + " WHERE table_schema = " + "'" + TEST_SCHEMA + "'";
+
+        assertQuery(countIcebergTableColumns, "VALUES 6");
```

Comment:
What's being tested here?

Reply:
OK. Add a comment: `// Verify information_schema.columns does not include columns from non-Iceberg tables`

Also, testing with `SELECT table_name, column_name. ..` and `VALUES ('iceberg_table1', '_string'), ...` will be more readable than just 6, especially when we need to update the test. (From looking at the code it's not obvious whether 6 is the correct expected value, so any change in this line will be hard to review).